clog/builder_test.go

```diff
@@ -0,0 +1,192 @@
+		assert.Equal(t, bld.with["foo"].(string), "bar")
+		assert.Contains(t, bld.with, "baz")
+		assert.Equal(t, bld.with["baz"].(int), 1)
+
+		bld.Label("l1", "l2", "l1")
+		assert.Contains(t, bld.labels, "l1")
+		assert.Contains(t, bld.labels, "l2")
+
+		bld.Comment("a comment")
+		bld.Comment("another comment")
+		bld.Comment("a comment")
+		assert.Contains(t, bld.comments, "a comment")
+		assert.Contains(t, bld.comments, "another comment")
+
+		bld.SkipCaller(1)
+
+		// ensure no collision between separate builders
+		// using the same ctx.
+		err := clues.New("an error").
+			With("fnords", "i have seen them").
+			Label("errLabel")
+
+		other := CtxErr(ctx, err)
+		assert.Empty(t, other.with)
+		assert.Empty(t, other.labels)
+		assert.Empty(t, other.comments)
+		assert.ErrorIs(t, other.err, err, clues.ToCore(err))
+
+		other.With("foo", "smarf")
+		assert.Contains(t, other.with, "foo")
+		assert.Equal(t, bld.with["foo"].(string), "bar")
+
+		other.Label("l3")
+		assert.Contains(t, other.labels, "l3")
+		assert.NotContains(t, bld.labels, "l3")
+
+		other.Comment("comment a")
+		assert.Contains(t, other.comments, "comment a")
+		assert.NotContains(t, bld.comments, "comment a")
+
+		// ensure no panics when logging
+		suite.testDebugLogs(bld)
+		suite.testInfoLogs(bld)
+		suite.testErrorLogs(bld)
+	}
+}
+
+func (suite *BuilderUnitSuite) testDebugLogs(bld *builder) {
+	bld.Debug("a", "log")
+	bld.Debugf("a %s", "log")
+	bld.Debugw("a log", "with key")
+	bld.Debugw("a log", "with key", "and value")
+	// negative skip caller, just to ensure safety
+	bld.
+		SkipCaller(-1).
+		Debugw("a log", "with key", "and value")
+}
+
+func (suite *BuilderUnitSuite) testInfoLogs(bld *builder) {
+	bld.Info("a", "log")
+	bld.Infof("a %s", "log")
+	bld.Infow("a log", "with key")
+	bld.Infow("a log", "with key", "and value")
+	// negative skip caller, just to ensure safety
+	bld.
+		SkipCaller(-1).
+		Infow("a log", "with key", "and value")
+}
+
+func (suite *BuilderUnitSuite) testErrorLogs(bld *builder) {
+	bld.Error("a", "log")
+	bld.Errorf("a %s", "log")
+	bld.Errorw("a log", "with key")
+	bld.Errorw("a log", "with key", "and value")
+	// negative skip caller, just to ensure safety
+	bld.
+		SkipCaller(-1).
+		Errorw("a log", "with key", "and value")
+}
+
+func (suite *BuilderUnitSuite) TestGetValue() {
+	table := []struct {
+		name     string
+		value    any
+		expected any
+	}{
+		{
+			name:     "integer",
+			value:    1,
+			expected: 1,
+		},
+		{
+			name:     "string",
+			value:    "foo",
+			expected: "foo",
+		},
+		{
+			name: "pointer to string",
+			value: func() *string {
+				s := "foo"
+				return &s
+			}(),
+			expected: "foo",
+		},
+		{
+			name:     "nil",
+			value:    nil,
+			expected: nil,
+		},
+		{
+			name: "nil pointer",
+			value: func() *string {
+				return nil
+			}(),
+			expected: nil,
+		},
+	}
+
+	for _, tt := range table {
+		suite.T().Run(tt.name, func(t *testing.T) {
+			assert.Equal(t, tt.expected, getValue(tt.value))
```

Comment:
I'm not actually sure if this will just end up testing the `interface` test case a bunch of time since that's the type in the struct or if it will be testing the underlying types

Reply:
we test both casses with this spread.  I'll add some other cases for coverage, just to have that clear.

Reply:
see: https://github.com/alcionai/clues/pull/69

Reply:
I think maybe we're on slightly different trains of thought here? My concern is whether declaring the test table struct to have `value any` for the input will result in always testing the interface segment of the code under test since technically `any` is an alias to `interface{}`. If that is the way the code behaves, then adding additional test cases to this test won't change anything

I think to figure out how this is actually run, you'd need to add temporary print statements to the code